src/Features/LanguageServer/Protocol/Features/Diagnostics/DefaultDiagnosticAnalyzerService.cs

```diff
@@ -75,23 +75,10 @@ public DefaultDiagnosticIncrementalAnalyzer(DefaultDiagnosticAnalyzerService ser
                 _service = service;
                 _workspace = workspace;
                 _diagnosticAnalyzerRunner = new InProcOrRemoteHostAnalyzerRunner(service._analyzerInfoCache);
-                _service._globalOptions.OptionChanged += OnGlobalOptionChanged;
             }
 
             public void Shutdown()
             {
-                _service._globalOptions.OptionChanged -= OnGlobalOptionChanged;
-            }
-
-            private void OnGlobalOptionChanged(object sender, OptionChangedEventArgs e)
-            {
-                if (e.Option == InternalRuntimeDiagnosticOptions.Syntax ||
-                    e.Option == InternalRuntimeDiagnosticOptions.Semantic ||
-                    e.Option == InternalRuntimeDiagnosticOptions.ScriptSemantic)
-                {
-                    var service = _workspace.Services.GetService<ISolutionCrawlerService>();
-                    service?.Reanalyze(_workspace, this, projectIds: null, documentIds: null, highPriority: false);
```

Comment:
>Reanalyze [](http://example.com/codeflow?start=29&length=9)

Shouldn't this be invoked somehow?

Reply:
we're never changing these options, so unclear why we'd need to even reanalyze.